lib/logger/lib/logger/filter.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Logger.Filter do
+  @moduledoc false
+
+  @doc """
+  Filter out SASL reports if `sasl_enabled?` is `true`
+  """
+  def sasl(sasl_enabled?) do
+    filter =
+      fn
+        %{meta: %{domain: [:otp, :sasl | _]}}, false -> :stop
+        %{meta: %{domain: [:supervisor_report | _]}}, false -> :stop
+        log, _sasl_enabled -> log
+      end
+
+    {filter, sasl_enabled?}
+  end
+
+  @doc """
+  Filter out logs if current process opted out of log reports
+  """
+  def process_disabled() do
+    {fn log, _extra ->
+      if Logger.enabled?(self()) do
```

Comment:
Is there a difference between adding the filters here or simply doing this check by the time we log? Otherwise, it would be nice to NOT use filters for our own stuff (and simlpy check by the time of logging) and completely leave filters for the users to manage.

Reply:
No there is not, except the fact that `process_disabled` filter is registered globally. `sasl` filter can be done within handler. The main difference is that if someone defines their own formatter then it will **not** run if any filter returns `:stop`. So there can be some problems when someone will expect formatter to not run on `sasl` entries.

Reply:
@hauleth so let's move the sasl check inside, since that's also how it works now (sasl is always checked inside). :+1: for leaving process_disabled out, great idea.

Reply:
Sounds good! Maybe we can even add upstream!

Reply:
@josevalim about `sasl` - should it be possible to change this option in the runtime? With `logger` it is not much of the problem.

Reply:
@hauleth if the goal is to remove the feature, than I wouldn't worry about improving it. :) I would just make it a filter that is added (or removed if false) on logger start.

Reply:
@josevalim after reviewing we should not remove that feature, as the default filtering of SASL reports depends on logging level (which is `notice` by default) as SASL messages are on `debug` level. So we need filtering anyway, but I can handle it within handler.

Also I have send proposal to add per-process log level on Erlang mailing list and I had already prepared commit with that feature - https://github.com/hauleth/otp/commit/77246b9323e26a74794153d36933773de0b83a6d

Reply:
erlang/otp#2384 is stalled, so we need to handle it on our own for now.